metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1096,106 +1117,74 @@ def get_tables_for_dataset(
-                    )
-                    table_name = table_identifier.get_table_name().split(".")[-1]
-
-                    # For sharded tables we only process the latest shard
-                    # which has the highest date in the table name.
-                    # Sharded tables look like: table_20220120
-                    # We only has one special case where the table name is a date
-                    # in this case we merge all these tables under dataset name as table name.
-                    # For example some_dataset.20220110 will be turned to some_dataset.some_dataset
-                    # It seems like there are some bigquery user who uses this way the tables.
-                    if shard:
-                        if not sharded_tables.get(table_identifier.get_table_name()):
-                            # When table is only a shard we use dataset_name as table_name
-                            sharded_tables[table_name] = table
-                            continue
-                        else:
-                            stored_table_identifier = BigqueryTableIdentifier(
-                                project_id=project_id,
-                                dataset=dataset_name,
-                                table=sharded_tables[table_name].table_id,
-                            )
-                            (
-                                _,
-                                stored_shard,
-                            ) = BigqueryTableIdentifier.get_table_and_shard(
-                                stored_table_identifier.raw_table_name()
-                            )
-                            # When table is none, we use dataset_name as table_name
-                            table_name = table_identifier.get_table_name().split(".")[
-                                -1
-                            ]
-                            assert stored_shard
-                            if stored_shard < shard:
-                                sharded_tables[table_name] = table
-                            continue
+                _, shard = BigqueryTableIdentifier.get_table_and_shard(
+                    table_identifier.raw_table_name()
+                )
+                table_name = table_identifier.get_table_name().split(".")[-1]
+
+                # For sharded tables we only process the latest shard
+                # which has the highest date in the table name.
+                # Sharded tables look like: table_20220120
+                # We only has one special case where the table name is a date
+                # in this case we merge all these tables under dataset name as table name.
+                # For example some_dataset.20220110 will be turned to some_dataset.some_dataset
+                # It seems like there are some bigquery user who uses this way the tables.
+                if shard:
+                    if not sharded_tables.get(table_identifier.get_table_name()):
```

Comment:
This seems dangerous. Why not use the existing table_name variable defined just above? We go on to update the map with the above variable as the key

Reply:
Fixed